storage/innobase/srv/srv0srv.cc

```diff
@@ -1486,6 +1528,21 @@ srv_export_innodb_status(void)
 
 	export_vars.innodb_buffer_pool_pages_misc =
 		buf_pool_get_n_pages() - LRU_len - free_len;
+	
+	export_vars.innodb_background_log_sync = srv_log_writes_and_flush;
+
+	export_vars.innodb_master_thread_active_loops = srv_main_active_loops;
+	export_vars.innodb_master_thread_idle_loops = srv_main_idle_loops;
+	export_vars.innodb_max_trx_id = trx_sys.get_max_trx_id();
+	export_vars.innodb_history_list_length = trx_sys.rseg_history_len;
+	export_vars.innodb_mem_dictionary = mem_dictionary;
+	export_vars.innodb_s_lock_os_waits = rw_lock_stats.rw_s_os_wait_count;
+	export_vars.innodb_s_lock_spin_rounds = rw_lock_stats.rw_s_spin_round_count;
+	export_vars.innodb_s_lock_spin_waits = rw_lock_stats.rw_s_spin_wait_count;
+	export_vars.innodb_x_lock_os_waits = rw_lock_stats.rw_x_os_wait_count;
+	export_vars.innodb_x_lock_spin_rounds = rw_lock_stats.rw_x_spin_round_count;
+	export_vars.innodb_x_lock_spin_waits = rw_lock_stats.rw_x_spin_wait_count;
+	export_vars.innodb_deadlocks = srv_stats.lock_deadlock_count;
```

Comment:
I would suggest removing most of these, and just binding the variables to `innodb_status_variables` directly.

Obviously this cannot be done for the `trx_sys` members, because they use `std::atomic`, or for the local variable `mem_dictionary`.

Reply:
I'm confused as to why we can't directly bind for `trx_sys` members that use atomic counters, because it seems okay to directly access them.
For instance, we currently have:
```
export_vars.innodb_max_trx_id = trx_sys.get_max_trx_id();
```
but `trx_sys.get_max_trx_id()` just directly returns `trx_sys.m_max_trx_id`:
```
trx_id_t get_max_trx_id()
  {
    return m_max_trx_id;
  }
```
So wouldn't it be okay to directly access  `trx_sys.m_max_trx_id` or are we assuming future changes to `trx_sys.get_max_trx_id()` will make it do more than just `return m_max_trx_id;`?

Reply:
@dr-m how do you bind them directly? Most of these are `Atomic_counter`s, which wrap `std::atomic`. Even though they're accessible directly (probably on all supported platforms), this is not something that C++11 advertises.

As for `m_max_trx_id` - it is subject for concurrent access. Having it private makes following its access patter much simpler.

Reply:
@svoj Sorry, I may have missed that many of the variables are `Atomic_counter`. Obviously we cannot directly bind anything that is wrapped by `std::atomic`, because that would likely be undefined behavior.